### PR TITLE
Added simplified version of MQAD

### PR DESCRIPTION
**Pull request by dkeeghan**
I've added a new function called `.at()` which is used in a similar way to our DD FED framework's Media Query Attach/Detach (MQAD) code.

This is a lot simpler than the MQAD code because that code allows for a choice of attaching or detaching to a breakpoint. We don't actually need to have that additional complexity by only allowing the user to attach to a breakpoint and assuming that anything that doesn't match the breakpoint fires the detach function instead.

This is a much simpler approach than we currently use and will help streamline some of our internal functions.

I do a check for enquire.js and issue a warning if the user doesn't have it included so you can still use the library without needing enquire if you don't want it.

### CODE REVIEW

**Comment by dkeeghan** [comment]
@keeganstreet @hadimichael Keen to get your thoughts guys. I'd like to get this merged in so I can use it in some upcoming updates to our internal JS plugins

**Comment by keeganstreet** [comment]
Hey, nice work. I think it makes sense to offer this event listener binding straight from the DDBreakpoints plugin.

I am wondering if we should bake in the enquire.js dependency, or maybe this would be a good time to move to using the native `matchMedia.addListener` API. The main advantage of enquire.js is that its easier for developers to use than the native API, but if its going to be abstracted away into DD Breakpoints, developers won't have to interact with the native API anyway. Paul Irish/Scott Jehl's matchMedia polyfill adds support for `matchMedia.addListener` in IE9.

**Comment by hdmchl** [comment]
TL;DR Awesome. Makes a lot of sense as a utility and provides a clean wrapper around enquire.

The purist in me is asking whether this should be part of the core library? Especially since to use it, you need a dependency (enquire.js). Would it be better to have another "utility" library on `window.DD`. For example:

``` javascript
...
window.DD = window.DD || {};

window.DD.at = function(bp, attach, detach) {
  //... some checks and warnings about how this lib has a dependency on DD.bp and enquire.js etc...

  if (DD.bp.is(bp)) {
    onAttach(true);
  } else {
    onDetach(true);
  }

  enquire.register(DD.bp.get(bp), {
    match: onAttach,
    unmatch: onDetach
  });
}
...
```

Benefits of this approach include:
- Keeping `DD.bp` tiny and minimal
- Making a start on open-sourcing `mqad`
- Showing that `DD.bp` is "pluggable"

**Comment by hdmchl** [comment]
Just saw @keeganstreet's comment. Agree that we should move to native APIs.

@dkeeghan: If you need to ship this quickly. Then creating a new "library" (let's say we call it `DD.at`) that has dependencies on `DD.bp` and `enquire.js` is a good first step. When we have time in the future, we can remove `DD.at`'s dependency on `enquire`, and implement native event listeners. Thoughts?

**Comment by dkeeghan** [comment]
I like @keeganstreet's comment around making it native. My original intention was to convert mqad into a jQuery plugin but when I looked at it, MQAD was around originally with much older versions of the breakpoints code and a lot of the complexity in it is now part of the `DD.bp` library instead (e.g. the ability to do min,max as a string, and the automatic figuring out if the breakpoint should work in non-responsive mode), so when I stripped out the unneeded stuff I realised that the code itself is very simple, and didn't really require an additional library.

I'm also thinking that MQAD doesn't need to be driven by the DOM anymore, you just implement it into the library that you want to use it instead.

**Comment by dkeeghan** [comment]
I'm going to close the pull request for the moment, implement a stopgap mqad plugin and we should investigate the native .addListener functionality without losing the support we've currently got for browsers less than IE8
